src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -378,6 +350,36 @@ public List<Course> getAllCoursesForManagementOverview(@RequestParam(defaultValu
         return courseService.getAllCoursesForManagementOverview(onlyActive);
     }
 
+    /**
+     * GET /courses/{courseId}/for-registration : get a single courses that the current user can register to.
+     * Decided by the start and end date and if the registrationEnabled flag is set correctly
+     *
+     * @param courseId the id of the course to retrieve
+     * @return the active course
+     */
+    @GetMapping("courses/{courseId}/for-registration")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Course> getCourseForRegistration(@PathVariable long courseId) {
+        log.debug("REST request to get a currently active course for registration");
+        User user = userRepository.getUserWithGroupsAndAuthoritiesAndOrganizations();
+        Course course = courseRepository.findSingleCurrentlyActiveNotOnlineAndRegistrationEnabledWithOrganizationsAndPrerequisitesElseThrow(courseId);
+
+        // check that the user can at least register to one organization of the course
+        if (!courseService.isUserAllowedToSelfRegisterForCourse(user, course)) {
+            throw new AccessForbiddenException("You are not allowed to register for this course, so you cannot see it.");
+        }
+
+        // check that the user has not already registered for the course
+        List<Course> allRegisteredCourses = courseService.findAllActiveForUser(user);
+        if (allRegisteredCourses.contains(course)) {
+            // redirect to the more detailed route /courses/{courseId}/for-dashboard
+            UriComponents redirectUri = UriComponentsBuilder.fromPath("/api/courses/{courseId}/for-dashboard").buildAndExpand(courseId);
+            return ResponseEntity.status(HttpStatus.FOUND).location(redirectUri.toUri()).build();
```

Comment:
This is a misuse of the HTTP 302.
The HTTP 302 specifies that a resource is available and was found but is at the moment not available under the expected address.

What actually happens is that the user doesn't have access (403) to the endpoint.

In my opinion, a better implementation would be to detect the 403 as a client and send a second request/redirect to a component that sends the request. If you then get another 4xx, it's clear that you are neither registered, not allowed to register.

I know this hits the core of your PR, but I think changes are necessary for the aforementioned reasons.

Please also remember that the 302 requires an implementation on the client side. While the browser implements this, any other API call would need to detect this and follow the redirect.

Reply:
I'm now using a `AccessForbiddenAlertException` instead, which is a much better solution 👍